services/server/config/config.go

```diff
@@ -41,6 +41,8 @@ type Config struct {
 	State string `toml:"state"`
 	// PluginDir is the directory for dynamic plugins to be stored
 	PluginDir string `toml:"plugin_dir"`
+	// WindowsPriorityClass is the priority class used to run containerd and child processes at on Windows machines
+	WindowsPriorityClass string `toml:"windows_priority_class"`
```

Comment:
This feels like a very OS-specific setting to have exposed on the top-level `Config` struct. Curious for other ideas but maybe putting it under some nested struct for OS-specific tuning?

```
type Config struct {
    ...
    OS OSConfig `toml:"os"`
    ...
}

type OSConfig struct {
    Windows WindowsConfig `toml:"windows"`
}

type WindowsConfig struct {
    PriorityClass string `toml:"priority_class"`
}
```

Reply:
I thought so too but the config does have `Cgroup` settings and `OOMScore`, so it's not alone here in the OS specific options lol. Would be nice to move those to the model you linked above but definitely not in this change

Reply:
Yea, i followed the same approach that the cgroup/OOMScore settings took here.

I can create a windows specific setting to add priority class under if people want.
Just let me know.

Reply:
Updated. I only added a Windows field to the OS options for now.
PTAL

Reply:
@containerd/reviewers any other opinions on the config changes here? If we expect to have other OS-specific settings in the future separating them out seems reasonable to me.

Reply:
> @containerd/reviewers any other opinions on the config changes here? If we expect to have other OS-specific settings in the future separating them out seems reasonable to me.

@containerd/reviewers - has anyone had a chance to think about this?